Site/SiteAnalyticsModule.php

```diff
@@ -221,9 +254,120 @@ public function getGoogleAnalyticsTrackerInlineJavascript()
 	}
 
 	// }}}
-	// {{{ protected function getTrackingCodeSource()
+	// {{{ public function hasFacebookPixel()
+
+	public function hasFacebookPixel()
+	{
+		return (
+			$this->facebook_pixel_id != '' &&
+			!$this->analytics_opt_out
+		);
+	}
+
+	// }}}
+	// {{{ public function pushFacebookPixelCommands()
+
+	public function pushFacebookPixelCommands(array $commands)
+	{
+		foreach ($commands as $command) {
+			$this->facebook_pixel_commands[] = $command;
+		}
+	}
+
+	// }}}
+	// {{{ public function prependFacebookPixelCommands()
+
+	public function prependFacebookPixelCommands(array $commands)
+	{
+		foreach ($commands as $command) {
+			array_unshift($this->facebook_pixel_commands, $command);
+		}
+	}
+
+	// }}}
+	// {{{ public function getFacebookPixelImage()
+
+	public function getFacebookPixelImage()
+	{
+		$xhtml = <<<'XHTML'
+<noscript><img height="1" width="1" style="display:none" src="https://www.facebook.com/tr?id=%s&ev=PageView&noscript=1"/></noscript>
+XHTML;
+
+		return sprintf(
+			$xhtml,
+			SwatString::minimizeEntities($this->facebook_pixel_id)
```

Comment:
Since this gets added as a GET var it should be `rawurlencod`ed before minimized.